packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -207,57 +223,74 @@ class GenerateLocalizationsCommand extends FlutterCommand {
-    final String? projectPathString = stringArgDeprecated('project-dir');
-    final bool areResourceAttributesRequired = boolArgDeprecated('required-resource-attributes');
-    final bool usesNullableGetter = boolArgDeprecated('nullable-getter');
+      precacheLanguageAndRegionTags();
 
-    precacheLanguageAndRegionTags();
+      try {
+        outputFileList = (LocalizationsGenerator(
+          fileSystem: _fileSystem,
+          inputPathString: inputPathString,
+          outputPathString: outputPathString,
+          templateArbFileName: templateArbFileName,
+          outputFileString: outputFileString,
+          classNameString: classNameString,
+          preferredSupportedLocales: preferredSupportedLocales,
+          headerString: headerString,
+          headerFile: headerFile,
+          useDeferredLoading: useDeferredLoading,
+          inputsAndOutputsListPath: inputsAndOutputsListPath,
+          useSyntheticPackage: useSyntheticPackage,
+          projectPathString: projectPathString,
+          areResourceAttributesRequired: areResourceAttributesRequired,
+          untranslatedMessagesFile: untranslatedMessagesFile,
+          usesNullableGetter: usesNullableGetter,
+          logger: _logger,
+        )
+          ..loadResources()
+          ..writeOutputFiles())
+          .outputFileList;
+      } on L10nException catch (e) {
+        throwToolExit(e.message);
+      }
+    }
 
-    try {
-      LocalizationsGenerator(
-        fileSystem: _fileSystem,
-        inputPathString: inputPathString,
-        outputPathString: outputPathString,
-        templateArbFileName: templateArbFileName,
-        outputFileString: outputFileString,
-        classNameString: classNameString,
-        preferredSupportedLocales: preferredSupportedLocales,
-        headerString: headerString,
-        headerFile: headerFile,
-        useDeferredLoading: useDeferredLoading,
-        inputsAndOutputsListPath: inputsAndOutputsListPath,
-        useSyntheticPackage: useSyntheticPackage,
-        projectPathString: projectPathString,
-        areResourceAttributesRequired: areResourceAttributesRequired,
-        untranslatedMessagesFile: untranslatedMessagesFile,
-        usesNullableGetter: usesNullableGetter,
-        logger: _logger,
-      )
-        ..loadResources()
-        ..writeOutputFiles();
-    } on L10nException catch (e) {
-      throwToolExit(e.message);
+    // All other post processing.
+    if (boolArg('format') ?? false) {
+      if (_artifacts == null) {
+        throwToolExit('Could not find artifacts');
+      }
+      if (outputFileList.isEmpty) {
+        return FlutterCommandResult.success();
+      }
+      final String dartBinary = _artifacts!.getHostArtifact(HostArtifact.engineDartBinary).path;
+      final List<String> command = <String>[dartBinary, 'format', ...outputFileList];
+      final Process process = await _processManager.start(command);
```

Comment:
If you don't need to interact with the sub-process, calling .run() is cleaner:

```dart
final ProcessResult result = await _processManager.run(command);
if (result.exitCode != 0) {
```

Reply:
Fixed!